SentinelHub/settings.py

```diff
@@ -61,47 +63,43 @@ class Settings:
         "lng_max",
         "download_folder",
     }
-    _auto_save = False
     _CREDENTIAL_STORE_PARAMETERS = {"base_url", "client_id", "client_secret"}
     _STORE_PARAMETERS = _AUTO_SAVE_STORE_PARAMETERS | _CREDENTIAL_STORE_PARAMETERS
 
-    def __init__(self):
+    def __init__(self, path: Optional[str] = None):
+        super().__init__(path)
         self.load_local_settings()
-        self._auto_save = True
 
-    def __setattr__(self, key, value):
+    def __setattr__(self, key: str, value: Any) -> None:
         """Whenever one of the attributes from _AUTO_SAVE_STORE_PARAMETERS is set it is automatically saved to
         QGIS store
         """
-        if self._auto_save and key in self._AUTO_SAVE_STORE_PARAMETERS:
+        if key in self._AUTO_SAVE_STORE_PARAMETERS:
             store_path = self._get_store_path(key)
-            QSettings().setValue(store_path, value)
-
+            self.setValue(store_path, value)
         super().__setattr__(key, value)
 
-    def load_local_settings(self):
-        """Loads settings from QGIS local store"""
-        qsettings = QSettings()
-
+    def load_local_settings(self) -> None:
         for parameter in self._STORE_PARAMETERS:
             store_path = self._get_store_path(parameter)
-            store_value = qsettings.value(store_path)
+            store_value = self.value(store_path)
 
             if store_value is not None:
                 setattr(self, parameter, str(store_value))
 
-    def save_credentials(self):
-        """Saves settings to QGIS local store"""
-        qsettings = QSettings()
+    @staticmethod
+    def _get_store_path(parameter: str) -> str:
+        return f"{Settings._STORE_NAMESPACE}/{parameter}"
 
+    def save_credentials(self) -> None:
+        """Saves settings to QGIS local store"""
         for parameter in self._CREDENTIAL_STORE_PARAMETERS:
             store_path = self._get_store_path(parameter)
-            qsettings.setValue(store_path, getattr(self, parameter))
-
-    def _get_store_path(self, parameter_name):
-        """Provides a location of the parameter in the local store"""
-        return f"{self._STORE_NAMESPACE}/{parameter_name}"
-
-    def copy(self):
-        """Provides a copy of a Settings object instance"""
-        return copy.copy(self)
+            self.setValue(store_path, getattr(self, parameter))
+
+    def copy(self) -> T:
```

Comment:
Here the type should be `Settings` due to L101. I don't think you should lose sleep over this, unlikely someone will subclass it further.

Reply:
fixed